### PR TITLE
Fix  #1057 issue for selection with mouse + gaze

### LINKED ISSUE #1057: Memory game error
When 2 cards are selected at the same time (by mouse + eye tracker) we can't continue the game. (No other flip possible and the 2 cards stay flipped)

### PR DESCRIPTION
**Pull request by kuhlkrein**
fix for #1057

### CODE REVIEW

**Comment by Tohaker** [comment]
Do you think you could write some tests for the changed code?

**Comment by kuhlkrein** [comment]
@Tohaker yes, it's mainly changes related to Timeline code sections, do you know what kind of test I could right for it ?

**Comment by Tohaker** [comment]
I'd suggest some unit tests that test the various way to returning from the method, by mocking the GameInstance. You should find plenty of example tests in the project already

**Comment by kuhlkrein** [comment]
Ok thank you @Tohaker  I'll try that :)

**Comment by YannRobert** [comment]
Maybe need to reword the PR title ?
and reference link to original issue if exists

File: gazeplay-games/src/test/java/net/gazeplay/games/memory/MemoryTest.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    @EnumSource(value = Memory.MemoryGameType.class, mode = EnumSource.Mode.EXCLUDE, names = {"CUSTOMIZED"})
+    void shouldPickAndBuildRandomPictures(Memory.MemoryGameType gameType) {
+        Memory memory = new Memory(gameType, mockGameContext, 2, 2, mockStats, true);
+        when(mockConfig.getLanguage()).thenReturn("eng");
+
+        Dimension2D mockDimension = new Dimension2D(20, 20);
+        when(mockGameContext.getGamePanelDimensionProvider().getDimension2D()).thenReturn(mockDimension);
+        when(mockGameContext.getConfiguration()).thenReturn(mockConfig);
+
+        HashMap<Integer, Image> randomPictures = memory.selectionAleaImages();
```
Comment (Tohaker) [inline gazeplay-games/src/test/java/net/gazeplay/games/memory/MemoryTest.java]: What does `selectionAleaImages` mean? Should it be `Area`?

Comment (kuhlkrein) [reply gazeplay-games/src/test/java/net/gazeplay/games/memory/MemoryTest.java]: Oh I didn't noticed, Alea is for Aleatoire, what means random in french, the name of the method means "random image selection", it should have been created by a french person lol, I will change it if you want :)

Comment (Tohaker) [reply gazeplay-games/src/test/java/net/gazeplay/games/memory/MemoryTest.java]: Ah I see, I think we should keep the language consistent within names so we don't confuse people, but good to know in case I see it again!

**Comment by kuhlkrein** [comment]
Need to revert modification of AbstractGazeDeviceManager.java before merging, this modification have to be done in #1059

File: gazeplay-commons/src/main/java/net/gazeplay/commons/gaze/devicemanager/AbstractGazeDeviceManager.java [hunk 2]
```diff
@@ -193,6 +195,13 @@ public void eventFire(double positionX, double positionY, GazeInfos gi, Node nod
         if (!node.isDisable()) {
 
             Point2D localPosition = node.screenToLocal(positionX, positionY);
+
+            Scene scene = node.getScene();
+            if (scene != null) {
+                Window window = scene.getWindow();
+                localPosition = localPosition.add(window.getX(), window.getY());
+            }
+
```
Comment (YannRobert) [inline gazeplay-commons/src/main/java/net/gazeplay/commons/gaze/devicemanager/AbstractGazeDeviceManager.java]: this change is not related to the other changes in the same PR
maybe you should isolate this change in a separate PR

Comment (kuhlkrein) [reply gazeplay-commons/src/main/java/net/gazeplay/commons/gaze/devicemanager/AbstractGazeDeviceManager.java]: Yes It's what I said in the last comment, I need to remove this file from the commit.

Comment (kuhlkrein) [reply gazeplay-commons/src/main/java/net/gazeplay/commons/gaze/devicemanager/AbstractGazeDeviceManager.java]: Done :)

**Comment by Tohaker** [comment]
@kuhlkrein could you resolve these conflicts?

**Comment by kuhlkrein** [comment]
I have to check few things before merging

**Comment by kuhlkrein** [comment]
ok, seems good, I checked everything and I should have not break the @YannRobert 's part. Just keeped the "final" declarations, and actually the rest was ok
